src/drivers/fluid_sdl3.c

```diff
@@ -0,0 +1,449 @@
+        if (!newlist.devices) {
+            SDL_free(devices);
+            return list->num_devices;  /* just return the existing one for now. Oh well. */
+        }
+
+        for (i = 0; i < num_devices; i++) {
+            const char *newname = SDL_GetAudioDeviceName(devices[i]);
+            char *fullname = NULL;
+            if (newname == NULL) {
+                /* ugh, whatever, just make up a name. */
+                newname = "Unidentified device";
+            }
+
+            /* Device names must be unique in sdl3, as that's how we open them.
+               sdl3 took serious pains to try to add numbers to the end of duplicate device names ("SoundBlaster Pro" and then "SoundBlaster Pro (2)"),
+               but here we're just putting the actual SDL3 instance id at the end of everything. Good enough. I hope. */
+            if (!newname || (SDL_asprintf(&fullname, "%s (id=%u)", newname, (unsigned int) devices[i]) < 0)) {
+                /* we're in real trouble now.  :/  */
+                int j;
+                for (j = 0; j < i; j++) {
+                    SDL_free(newlist.devices[i].name);
+                }
+                SDL_free(fullname);
+                SDL_free(devices);
+                return list->num_devices;  /* just return the existing one for now. Oh well. */
+            }
+
+            newlist.devices[i].devid = devices[i];
+            newlist.devices[i].name = fullname;
+        }
+    }
+
+    if (list != NULL)
+    {
+        for (i = 0; i < list->num_devices; i++) {
+            SDL_free(list->devices[i].name);
+        }
+        SDL_free(list->devices);
+    }
+
+    SDL_free(devices);
+
+    if (list != NULL)
+    {
+        SDL_memcpy(list, &newlist, sizeof (AudioDeviceList));
+    }
+
+    return num_devices;
+}
+
+static void
+SDLAudioCallback(void *data, SDL_AudioStream *stream, int add_len, int len)
+{
+    void *buffer = malloc(len);
+    fluid_sdl3_audio_driver_t *dev = (fluid_sdl3_audio_driver_t *)data;
+    (void)add_len;
+
+    if (buffer == NULL)
+    {
+        return;
+    }
+
+    len /= dev->frame_size;
+
+    dev->write_ptr(dev->synth, len, buffer, 0, 2, buffer, 1, 2);
+    SDL_PutAudioStreamData(dev->stream, buffer, len);
+
+    if (buffer != NULL)
+    {
+        free(buffer);
+    }
+}
+
+void fluid_sdl3_audio_driver_settings(fluid_settings_t *settings)
+{
+    int n = 0, j = 0, nDevs = 0;
+    SDL_Mutex *AudioDeviceLock = SDL_CreateMutex();
+    AudioDeviceList *list = NULL;
+    SDL_AudioDeviceID *devices = NULL;
+
+    fluid_settings_register_str(settings, "audio.sdl3.device", "default", 0);
+    fluid_settings_add_option(settings, "audio.sdl3.device", "default");
+
+    if(!SDL_WasInit(SDL_INIT_AUDIO))
+    {
+#if FLUID_VERSION_CHECK(FLUIDSYNTH_VERSION_MAJOR, FLUIDSYNTH_VERSION_MINOR, FLUIDSYNTH_VERSION_MICRO) < FLUID_VERSION_CHECK(2,2,0)
+        FLUID_LOG(FLUID_WARN, "SDL3 not initialized, SDL3 audio driver won't be usable");
+#endif
```

Comment:
```suggestion
        FLUID_LOG(FLUID_WARN, "SDL3 not initialized, SDL3 audio driver won't be usable");
```

This version check is used to prevent log messages when using an old fluidsynth executable and a recent libfluidsynth. I don't think it's needed here, so we can safely remove it.

Reply:
Removed version checks